.github/workflows/run_js_test.yml

```diff
@@ -37,7 +37,7 @@ jobs:
 
     - name: Upload `coverage` folder to R2
       if: ${{ always() }}
-      uses: ryand56/r2-upload-action@latest
+      uses: himanshu-dixit/r2-upload-action-parallel@v1
```

Comment:
While changing to a parallel upload action is a good optimization, there are a few concerns:

1. The action change is only applied to the coverage folder upload but not to the HTML report upload (line 56). For consistency and optimal performance, consider updating both upload steps to use the parallel version.

2. Using a specific version (`@v1`) is better than `@latest`, but please add:
   - Link to the action's source code repository
   - Brief comment explaining the performance benefits
   - Consider documenting any benchmarks/improvements seen

3. Consider adding error handling or fallback mechanism in case the parallel upload fails.